platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/collections/LongListDisk.java

```diff
@@ -16,22 +16,76 @@
 
 package com.swirlds.merkledb.collections;
 
+import com.swirlds.common.io.utility.TemporaryFileBuilder;
 import com.swirlds.merkledb.utilities.MerkleDbFileUtils;
 import java.io.IOException;
+import java.io.RandomAccessFile;
 import java.io.UncheckedIOException;
 import java.nio.ByteBuffer;
 import java.nio.ByteOrder;
 import java.nio.channels.FileChannel;
 import java.nio.file.Path;
 import java.nio.file.StandardOpenOption;
+import java.util.Deque;
+import java.util.List;
+import java.util.concurrent.ConcurrentLinkedDeque;
+import java.util.concurrent.CopyOnWriteArrayList;
 
-/** A direct on disk implementation of LongList */
-public class LongListDisk extends LongList {
+/**
+ *  A direct on disk implementation of LongList. This implementation creates a temporary file to store the data.
+ *  If the user provides a file, it will be used to create a copy of the in the temporary file. If the file is absent,
+ *  it will take the name of the file provided by the user and create a temporary file with this name.
+ * <p>
+ *  Unlike the "snapshot" file, the temporary files doesn't contain the header, only the body.
+ *
+ */
+public class LongListDisk extends LongList<Long> {
+
+    private static final String STORE_POSTFIX = "longListDisk";
+    private static final String DEFAULT_FILE_NAME = "LongListDisk.ll";
     /** A temp byte buffer for reading and writing longs */
-    private static final ThreadLocal<ByteBuffer> TEMP_LONG_BUFFER_THREAD_LOCAL =
-            ThreadLocal.withInitial(() -> ByteBuffer.allocateDirect(Long.BYTES).order(ByteOrder.nativeOrder()));
-    /** The disk file this LongList is based on */
-    private final Path file;
+    private final ThreadLocal<ByteBuffer> tempLongBufferThreadLocal;
```

Comment:
A non-static `ThreadLocal` should have a very good justification for its existence. 
These `tempLongBuffers` may potentially have different sizes (and they probably shouldn't). Is that a good justification?

Reply:
`tempLongBuffers` shouldn't have different sizes. But for `transferBufferThreadLocal` it can be the case.